src/server/error/parking-lot.service.ts

```diff
@@ -0,0 +1,7 @@
+export class ParkingLotNotFound extends Error {
+  constructor(criteria: string, value: string) {
+    super(`No parking lot with "${criteria}": ${value}, found`);
+
+    this.name = 'ParkingLotNotFound';
+  }
+}
```

Comment:
Whats the use case we are trying to cover with this? For instance, if we are throwing this error when a `ParkingLot` with an ID "X" is not found, then I rather having a `ParkingLotIDNotFound` or `UnexistentParkingLot`.

I would have the same error thrown in a function where Im searching for two parking lot entries, one with "name" criteria and another with "id" criteria, which invalidates the use of `instanceof`.

Consider this code snippet, where we rely on this error for handling the use case error:

```ts
// -- snip --
try {
    const parkingLotByName = await services.parkingLot.findByName('foo');
    const parkingLotById = await services.parkingLot.findById('123');
    // -- snip --
} catch (error) {
    if (error instanceof ParkingLotNotFound) {
        // how do we know which of both failed?
        // we probably would have to rely on messages
        // which is error prone and dirty because we could have
        // anything in a message at this point
        if (error.message === `No parking lot with "name": ${name_provided_to_function}, found`) {
            // do something with error
        }

        if (error.message === `No parking lot with "id": ${id_provided_to_function}, found`) {
           // do something with error
        }
    }
}
// -- snip --
```

I suggest having a dedicated error for this instead, if we are covering ID not found errors, then a `UnexistentParkingLotID` for instance